content/6_Advanced/String_Search.mdx

```diff
@@ -175,9 +175,7 @@ We would then have these two cases:
 #include <bits/stdc++.h>
 using namespace std;
 
-int main() {
-	string s;
-	cin >> s;
+vector<int> z_function(const string &s) {
```

Comment:
are we defining `z[0]` as being `n` or `0`? cuz the python one defines it as `n` (which I think makes more sense, IMO)